DQMOffline/Trigger/plugins/BPHMonitor.cc

```diff
@@ -658,25 +688,31 @@ void BPHMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
           break;    
   
         case 7:
-
-          if (!phHandle->empty()) for (auto const & p : *phHandle) {
-	      
-	      if ( !matchToTrigger(hltpath,p) ) continue;
-
-	      phPhi_.denominator->Fill(p.phi());
-	      phEta_.denominator->Fill(p.eta());
-	      phPt_.denominator ->Fill(p.pt());
-
-	      if ( num_genTriggerEventFlag_->on() && num_genTriggerEventFlag_->accept( iEvent, iSetup) ) {
-		if ( !matchToTrigger(hltpath1,p) ) continue;
-		if ( !matchToTrigger(hltpath1,m) ) continue;
-		if ( !matchToTrigger(hltpath1,m1) ) continue;
-		phPhi_.numerator->Fill(p.phi(),PrescaleWeight);
-		phEta_.numerator->Fill(p.eta(),PrescaleWeight);
-		phPt_.numerator ->Fill(p.pt(),PrescaleWeight);
+	  
+	  if ( phHandle.isValid() ) {
+	    if (!phHandle->empty()) for (auto const & p : *phHandle) {
+		
+		if ( !matchToTrigger(hltpath,p) ) continue;
+		
+		phPhi_.denominator->Fill(p.phi());
+		phEta_.denominator->Fill(p.eta());
+		phPt_.denominator ->Fill(p.pt());
+		
+		if ( num_genTriggerEventFlag_->on() && num_genTriggerEventFlag_->accept( iEvent, iSetup) ) {
+		  if ( !matchToTrigger(hltpath1,p) ) continue;
+		  if ( !matchToTrigger(hltpath1,m) ) continue;
+		  if ( !matchToTrigger(hltpath1,m1) ) continue;
+		  phPhi_.numerator->Fill(p.phi(),PrescaleWeight);
+		  phEta_.numerator->Fill(p.eta(),PrescaleWeight);
+		  phPt_.numerator ->Fill(p.pt(),PrescaleWeight);
+		}		
 	      }
-              
+	  } else {
+	    if (!warningPrinted4token_[3]) {
+	      edm::LogWarning("BPHMonitor") << "skipping events because the collection " << phInputTag_.label().c_str() << " is not available";
+	      warningPrinted4token_[3] = true;
```

Comment:
@mtosi don't you need a return statement here?

Reply:
sure, the code could still run, but it will give different results wrt the fully functional version. So depending on which input is missing you could get different results from the same code without immediately realising it (in the DQM checks, people will not normally look at log files).
It is a matter of choices, having no filled histogram is in my opinion perhaps preferable than having filled histograms filled in an unexpected way. But I think this is ultimately something that should be agreed by the end users. We are always speaking of pathological situations.